src/hotspot/os/aix/os_aix.cpp

```diff
@@ -1596,8 +1606,11 @@ static char* reserve_shmated_memory (size_t bytes, char* requested_addr) {
   memset(&shmbuf, 0, sizeof(shmbuf));
   shmbuf.shm_pagesize = 64*K;
   if (shmctl(shmid, SHM_PAGESIZE, &shmbuf) != 0) {
-    trcVerbose("Failed to set page size (need " UINTX_FORMAT " 64K pages) - shmctl failed with %d.",
-               size / (64*K), errno);
+    ErrorPreserver ep;
+    log_trace(os, map)("Failed to set page size (need " UINTX_FORMAT
+                       " 64K pages) - shmctl failed. (errno=%s).",
+                       size / (64 * K),
+                       os::strerror(ep.saved_errno()));
     // I want to know if this ever happens.
     assert(false, "failed to set page size for shmat");
```

Comment:
All of these messages could be printed with fatal() rather than assert(false) ?  Unless it's against some AIX policy?

Reply:
assert will only fire on debug builds whereas fatal will fire on all builds. I assume the intent is to catch this during testing but only log in production ??

Reply:
Let's use assert here. 

I think the assert vs guarantee/fatal question in these sources is a bit messy and could do with a do-over, but in a different PR and done by people that can test on AIX (neither I nor Sonia can).

Most of this code was done by me, 20 years ago. When we did the AIX port, initially things were broken left and right, and some AIX sysV APIs failed intermittently too, so I often used guarantee to make sure we noticed when this happened, even at a customer site. That said, all issues OS- and JVM-wise are long ironed out today, so assert is more appropiate.

Reply:
Ok. yes. then I assume this could be:

assert(false, "Failed to set page size (need " UINTX_FORMAT
                       " 64K pages) - shmctl failed. (errno=%s).",
                       size / (64 * K),
                       os::strerror(ep.saved_errno());

Or just leave it if you can't use fatal.